packages/cosmic-swingset/lib/ag-solo/vats/ibc.js

```diff
@@ -160,7 +216,13 @@ export function makeIBCProtocolHandler(
       },
       onReceive,
       async onClose(_conn, _reason, _handler) {
-        await callIBCDevice('channelCloseInit', { channelID, portID });
+        const packet = {
+          source_port: portID,
+          source_channel: channelID,
+        };
+        await callIBCDevice('channelCloseInit', { packet });
+        usedChannels.delete(channelID);
```

Comment:
Let's look carefully at this.. there's a danger of the two sides disagreeing about whether the channel is closed or not, and re-using channel IDs could be a security hole. TCP attempts to deal with this with timeouts. At the very least this shouldn't happen until `channelCloseConfirm`, but it might be safer to just never re-use channel IDs (I seem to remember IBC itself having some rule like this, but maybe it was on ClientIDs instead).

Reply:
Commented as such.